### PR TITLE
Enable PID file creation on WIN

### PR DESCRIPTION
**Pull request by riordant**
# Introduction 

As discussed with @laanwj on IRC: 

- PID file creation was never enabled for Windows, as the `pid_t` filetype is not available for it. However, the WIN32 API contains the header [`Processthreadsapi.h`](https://github.com/CodeShark/x86_64-w64-mingw32/blob/master/include/processthreadsapi.h) which in turn contains the function [`GetCurrentProcessId()`](https://docs.microsoft.com/en-gb/windows/desktop/api/processthreadsapi/nf-processthreadsapi-getcurrentprocessid). ~~This function is called at a higher level by [`_getpid()`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getpid?view=vs-2017)~~ EDIT: `_getpid()` is not available to the MSVC compiler used in the AppVeyor build. As a result, I have changed the function call to`GetCurrentProcessId()`, which performs the same function and is available to both MinGW & MSVC.
This allows one to capture the PID in Windows, without any additional includes - the above function is already available.

- Within this PR, I have added a separate line that calls `GetCurrentProcessId()` in the case of a WIN compilation, and the usual `getpid()` otherwise. All code blocks processing PID file logic that avoid WIN32 have been changed to consider it. I have also updated the preprocessor definitions in `libbitcoin_server.vcxproj.in` to suppress a warning related to `std::strerror` for the MSVC build, that was causing the AppVeyor build to fail (see @fanquake comment below).

# Rationale
- Consistency between OS's running Bitcoin
    - Applications which build off of `bitcoind`, such as novel front-end clients, often need access to the PID in order to control the daemon. Instead of designing some alternate way of doing this for one system, it should be consistent between all of them.

In collaboration with @joernroeder

### CODE REVIEW

**Comment by laanwj** [comment]
`+4 -10` and *adds* a feature, this is a great first contribution, thanks!
utACK ca33ac10d36199395bd6553db77e5ceacfdb925e

**Comment by riordant** [comment]
Force pushed in order to add co-author.

**Review by ken2812221 — APPROVED** [review]
~utACK c4372b8b18d2fab2101d904289b78f87703bb6b9~
You might have to include process.h?

**Comment by fanquake** [comment]
[Appveyor build](https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/22561871) is failing with:
```
c:\projects\bitcoin\src\init.cpp(112): error C3861: '_getpid': identifier not found [C:\projects\bitcoin\build_msvc\libbitcoin_server\libbitcoin_server.vcxproj]

c:\projects\bitcoin\src\init.cpp(119): error C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\projects\bitcoin\build_msvc\libbitcoin_server\libbitcoin_server.vcxproj]
```

**Comment by fanquake** [comment]
I compiled this PR inside WSL and ran `bitcoin-qt` on a Windows 10 system. 
Looks like `bitcoin.pid` is created correctly, and contains the correct PID.

windows

**Comment by riordant** [comment]
Update: AppVeyor build should now pass. I have updated the description with details

File: src/init.cpp [hunk 1]
```diff
@@ -286,15 +288,13 @@ void Shutdown(InitInterfaces& interfaces)
     }
 #endif
 
-#ifndef WIN32
     try {
         if (!fs::remove(GetPidFile())) {
             LogPrintf("%s: Unable to remove PID file: File does not exist\n", __func__);
         }
     } catch (const fs::filesystem_error& e) {
         LogPrintf("%s: Unable to remove PID file: %s\n", __func__, e.what());
```
Comment (ken2812221) [inline src/init.cpp]: nit: Use `fsbridge::get_filesystem_error_message(e)` instead of `e.what()`

Comment (riordant) [reply src/init.cpp]: Ok, can be added. Just note this was not added/modified by me and was merged in https://github.com/bitcoin/bitcoin/pull/15278 a few days ago

**Comment by practicalswift** [comment]
Concept ACK

Excellent first-time contribution! Thanks! Hope to see more PR:s from you! :-)

File: src/init.cpp (hunk 1 above)
Comment (ken2812221) [reply src/init.cpp]: That was OK to use `e.what()` in other OS except Windows. So you have to use `fsbridge::get_filesystem_error_message(e)` instead.

**Comment by hebasto** [comment]
Concept ACK.

**Comment by riordant** [comment]
Travis build failed at `feature_fee_estimation.py` line `116`. Seems completely unrelated?

**Comment by laanwj** [comment]
utACK after squash, I think this is better as one commit to have an atomic change

**Comment by riordant** [comment]
Yep agree. How can I do this?

**Comment by riordant** [comment]
Yep agree, done.

**Comment by laanwj** [comment]
thanks!
utACK 3f5ad622e5fe0781a70bee9e3322b23c2352e956

**Pull request merged**